### ISSUE TITLE
Some confuse about the definition of "bordermask"

### ISSUE THREAD

**Issue by EnQing626** [bryanyzhu/Hidden-Two-Stream#13]
Hello Yi, 
Thanks for your fantastic work and code!
I am new to the task of video and I have a question about the definition of "bordermask":
1. What is the DummyData layer about "flowBorderMask" and "borderMask" work for? 
2. Why do we need to use these masks when computing CharbonnierLoss?

Thanks so much in advance

**Comment by bryanyzhu** [comment]
Hi, thanks for your interest. The reason is because when you do the image reconstruction, the pixels at the border are usually wrong, maybe some pixels are already out of border or occluded. Hence, it is better to exclude these regions when computing the CharbonnierLoss. The training will be much smoother. Otherwise, the training loss curve will look oscillating because there are always many errornous pixels. It is hard for you to determine when to stop. 

The bordermask is something like a matrix as below: 

000000
0111110
0111110
000000

The details are defined in [here](https://github.com/bryanyzhu/Hidden-Two-Stream/blob/fdeb787082e9c4c0f8a8d2b2c9f22b73f7df2960/include/caffe/filler.hpp), you can check them out. Hope this is clear.

**Comment by EnQing626** [comment]
Very thanks for your help. It seems like a "padding" operation which sets 0 inside the image?

**Comment by bryanyzhu** [comment]
Yes, sets 0 around the border of the images, so we don't care how reconstruction performs in this area. 

The details can be seen [here](https://github.com/bryanyzhu/Hidden-Two-Stream/blob/fdeb787082e9c4c0f8a8d2b2c9f22b73f7df2960/include/caffe/filler.hpp#L82-L99). So we  first compute borderWidth given the ratio, for example 20 pixels. Then we set the border to 0. Both on the height and width dimension, we set 0 till 20 to 0, and w-20 till w to 0.
